src/ast/semantic_analyser.cpp

```diff
@@ -1710,5 +1714,36 @@ void SemanticAnalyser::assign_map_type(const Map &map, const SizedType &type)
   }
 }
 
+bool SemanticAnalyser::is_context(Expression *expr)
+{
+  if (auto *p = dynamic_cast<Builtin*>(expr))
```

Comment:
Doesn't have to be addressed now but we need to think harder about how to propagate inferred information "up" the AST. What we really want to know here is if `expr` requires a prog context. Doing these downcasts is kinda hacky b/c it lets things like this through:
```
$ sudo ./build/src/bpftrace --include linux/bpf_perf_event.h -e 'software:cpu:100000000 { printf("%d\n", ((struct bpf_perf_event_data*)((!ctx)+100))->sample_period); exit(); }'
[sudo] password for dxu:
Attaching 1 probe...
0
```